scope.c

```diff
@@ -1348,7 +1348,12 @@ Perl_leave_scope(pTHX_ I32 base)
 	    break;
 
 	case SAVEt_HINTS:
+        {
+	    HV *was_hinthv;
             a0 = ap[0]; a1 = ap[1];
+            if (a0.any_i32 & HINT_LOCALIZE_HH) {
```

Comment:
I would suggest moving that `if` block just one line after, as we are only going to use `was_hinthv` is `PL_hints & HINT_LOCALIZE_HH`
otherwise this looks good

Reply:
Nope, the point is that we have to `SSPOPPTR` which applies a globally-stateful change *before* we run any other code. Such as side-effects of `free` magic run as a result of `SvREFCNT_dec`ing here.

Reply:
Really, the true bug here is the non-constant stack shape of `SAVEt_HINTS` itself - which I aim to fix in #17895. This code here is currently very subtle because it contains the one lone instance of an `SSPOP*` macro in `Perl_leave_scope`, which is otherwise clean without them.

Reply:
thanks for your answer